examples/sp/sp_tutorial.py

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""A simple tutorial that shows some features of the Spatial Pooler."""
+
+import numpy as np
+import random
+import matplotlib.pyplot as plt
+from nupic.research.spatial_pooler import SpatialPooler as SP
+
+def pctOverlap(x1, x2, size):
+  """
+  Computes the percentage of overlap between vectors x1 and x2.
+  Parameters:
+  ----------
+  x1: binary vector whose size is specified in parameter size
```

Comment:
We use `@param` to document parameters (picked up by automated documentation tools). See nupic/src/nupic/research/temporal_memory.py for an example.